core/templates/domain/admin/admin-backend-api.service.ts

```diff
@@ -140,8 +140,70 @@ export class AdminBackendApiService {
       });
     });
   }
-}
+  flushCache():Promise<void> {
+    return this.http.post<void>(
+      AdminPageConstants.ADMIN_MEMORY_CACHE_HANDLER_URL, {}
+    ).toPromise();
+  }
+
+  clearSearchIndex():Promise<void> {
+    return this.http.post<void>(
+      AdminPageConstants.ADMIN_HANDLER_URL, {}
+    ).toPromise();
+  }
+
+  regenerateTopicRelatedOpportunities(topicId:string):Promise<void> {
+    return this.http.post<void>(
+      AdminPageConstants.ADMIN_HANDLER_URL, {
+        action: 'regenerate_topic_related_opportunities',
+        topic_id: topicId
+      }
+    ).toPromise();
+  }
+
+  uploadTopicSimilarities(data:string | ArrayBuffer):Promise<void> {
+    return this.http.post<void>(
+      AdminPageConstants.ADMIN_HANDLER_URL, {
+        action: 'upload_topic_similarities',
+        data: data,
+      }
+    ).toPromise();
+  }
 
+  sendDummyMail():Promise<void> {
+    return this.http.post<void>(
+      AdminPageConstants.ADMIN_SEND_DUMMY_MAIL_HANDLER_URL, {}
+    ).toPromise();
+  }
+
+  getMemoryCacheProfile():Promise<void> {
+    let memoryCacheProfile = null;
+    return new Promise((resolve, reject) => {
+      this.http.get<void>(
+        AdminPageConstants.ADMIN_MEMORY_CACHE_HANDLER_URL, {}
+      ).toPromise()
+        .then(responce =>{
+          memoryCacheProfile = responce;
```

Comment:
Undefined `memoryCacheProfile`? I'm expecting the test to fail, I'm not sure what's going on here!

Reply:
done!
![Screenshot from 2020-12-24 19-38-39](https://user-images.githubusercontent.com/54100702/103095253-f6021d00-4625-11eb-8b4b-d81583898f69.png)